enterprise/frontend/src/embedding-sdk/cli/utils/get-component-snippets.ts

```diff
@@ -26,33 +35,44 @@ export function getComponentSnippets(options: Options) {
     ? ANALYTICS_PROVIDER_SNIPPET_WITH_USER_SWITCHER
     : ANALYTICS_PROVIDER_SNIPPET_MINIMAL;
 
-  const components = [
+  const components: SampleComponent[] = [
     {
-      name: "analytics-provider",
-      content: analyticsProviderSnippet.trim(),
+      fileName: "analytics-provider",
+      componentName: "AnalyticsProvider",
+      content: withNextJsDirective(analyticsProviderSnippet, isNextJs).trim(),
     },
     {
-      name: "embedding-provider",
+      fileName: "embedding-provider",
+      componentName: "EmbeddingProvider",
       content: embeddingProviderSnippet.trim(),
     },
     {
-      name: "analytics-dashboard",
-      content: analyticsDashboardSnippet.trim(),
+      fileName: "analytics-dashboard",
+      componentName: "AnalyticsDashboard",
+      content: withNextJsDirective(analyticsDashboardSnippet, isNextJs).trim(),
```

Comment:
We use a `useReducer` here to switch between dashboards. Again we need the `"use client"` directive.